subliminal/providers/subscenter.py

```diff
@@ -0,0 +1,180 @@
+        if self.logged_in:
+            r = self.session.get(self.server + 'subscenter/accounts/logout/', timeout=10)
+            logger.info('Logged out')
+            if r.status_code != 200:
+                raise ProviderError('Request failed with status code %d' % r.status_code)
+            self.logged_in = False
+        self.session.close()
+
+    @staticmethod
+    def slugify(string):
+        new_string = string.replace(' ', '-').replace("'", '').replace(':', '').lower()
+        # We remove multiple spaces by using this regular expression.
+        return re.sub('-+', '-', new_string)
+
+    def query(self, languages=None, series=None, season=None, episode=None, title=None):
+        # Converts the title to Subscenter format by replacing whitespaces and removing specific chars.
+        if series and season and episode:
+            # Search for a TV show.
+            kind = 'episode'
+            slugified_series = self.slugify(series)
+            url = self.server + 'cinemast/data/series/sb/' + slugified_series + '/' + str(season) + '/' + \
+                str(episode) + '/'
+            page_link = self.server + 'subtitle/series/' + slugified_series + '/' + str(season) + '/' + \
+                str(episode) + '/'
+        elif title:
+            # Search for a movie.
+            kind = 'movie'
+            slugified_title = self.slugify(title)
+            url = self.server + 'cinemast/data/movie/sb/' + slugified_title + '/'
+            page_link = self.server + 'subtitle/movie/' + slugified_title + '/'
+        else:
+            raise ValueError('One or more parameters are missing')
+        logger.debug('Searching subtitles %r', {'title': title, 'season': season, 'episode': episode})
+        response = self.session.get(url)
+        if response.status_code != 200:
+            raise ProviderError('Request failed with status code %d' % response.status_code)
+
+        subtitles = []
+        response_json = json.loads(response.content.decode('UTF-8'))
+        for lang, lang_json in response_json.items():
+            lang_obj = Language.fromalpha2(lang)
+            if lang_obj in self.languages and lang in languages:
+                for group_data in lang_json.values():
+                    for quality in group_data.values():
+                        for sub in quality.values():
+                            release = sub.get('subtitle_version')
+                            download_link = self.server + 'subtitle/download/' + lang + '/' + str(sub.get('id')) + \
+                                '/?v=' + release + '&key=' + str(sub.get('key'))
+                            subtitles.append(SubsCenterSubtitle(lang_obj, series, season,
+                                                                episode, title, release, kind,
+                                                                bool(sub.get('hearing_impaired', 0)),
+                                                                download_link, page_link))
```

Comment:
The goal of the SubsCenterSubtitle class is to store all subtitle related stuff as given by the provider. So here you want to store anything that constitutes the download_link without duplicating information. Variables `lang`, `id`, `release` and `key` need to be stored. Not the download link.
You will reconstitute the download link in the `download` method using the information previously stored.